contrib/format-daffodil/src/main/java/org/apache/drill/exec/store/daffodil/DaffodilMessageParser.java

```diff
@@ -0,0 +1,196 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.drill.exec.store.daffodil;
+
+import org.apache.daffodil.japi.DataProcessor;
+import org.apache.daffodil.japi.Diagnostic;
+import org.apache.daffodil.japi.ParseResult;
+import org.apache.daffodil.japi.infoset.InfosetOutputter;
+import org.apache.daffodil.japi.io.InputSourceDataInputStream;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.InputStream;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * DFDL Daffodil Streaming message parser
+ * <p/>
+ * You construct this providing a DataProcessor obtained from the DaffodilDataProcessorFactory. The
+ * DataProcessor contains the compiled DFDL schema, ready to use, as well as whether validation
+ * while parsing has been requested.
+ * <p/>
+ * The DataProcessor object may be shared/reused by multiple threads each of which has its own copy
+ * of this class. This object is, however, stateful, and must not be shared by multiple threads.
+ * <p/>
+ * You must call setInputStream, and setInfosetOutputter before you call parse(). The input stream
+ * and the InfosetOutputter objects are also private to one thread and are stateful and owned by
+ * this object. Once you have called setInputStream, you should view the input stream as the private
+ * property of this object. The parse() will invoke the InfosetOutputter's methods to deliver parsed
+ * data, and it may optionally create diagnostics (obtained via getDiagnostics) indicating which
+ * kind they are via the getIsProcessingError, getIsValidationError.
+ * <p/>
+ * Note that the InfosetOutputter may be called many times before a processing error is detected, as
+ * Daffodil delivers result data incrementally.
+ * <p/>
+ * Validation errors do not affect the InfosetOutputter output calls, but indicate that data was
+ * detected that is invalid.
+ * <p/>
+ * When parse() returns, the parse has ended and one can check for errors/diagnostics. One can call
+ * parse() again if there is still data to consume, which is checked via the isEOF() method.
+ * <p/>
+ * There are no guarantees about where the input stream is positioned between parse() calls. In
+ * particular, it may not be positioned at the start of the next message, as Daffodil may have
+ * pre-fetched additional bytes from the input stream which it found are not part of the current
+ * infoset, but the next one. The positioning of the input stream may in fact be somewhere in the
+ * middle of a byte, as Daffodil does not require messages to be of lengths that are in whole byte
+ * units. Hence, once you give the input stream to this object via setInputStream, that input stream
+ * is owned privately by this class for ever after.
+ */
+public class DaffodilMessageParser {
+
+  private static final Logger logger = LoggerFactory.getLogger(DaffodilMessageParser.class);
```

Comment:
This logger needs to be Drill's logger for log messages originating in the drill bit.